Library/Homebrew/download_strategy.rb

```diff
@@ -406,18 +406,27 @@ def fetch(timeout: nil)
 
       ohai "Downloading #{url}"
 
-      resolved_url, _, url_time, _, is_redirection =
-        resolve_url_basename_time_file_size(url, timeout: end_time&.remaining!)
+      fresh =
+        if version.respond_to?(:latest?)
+          !version.latest?
+        else
+          true
+        end
+
+      resolved_url, _, last_modified, _, is_redirection =
+        begin
+          resolve_url_basename_time_file_size(url, timeout: end_time&.remaining!)
+        rescue ErrorDuringExecution
+          raise unless cached_location.exist? && fresh # rubocop:disable Style/UnlessLogicalOperators
+
+          [nil, nil, nil, nil, nil]
+        end
```

Comment:
```suggestion
      resolved_url, _, last_modified, _, is_redirection = begin
        resolve_url_basename_time_file_size(url, timeout: end_time&.remaining!)
      rescue ErrorDuringExecution
        raise if !cached_location.exist? || !fresh
      end
```

Reply:
Are you sure the `[nil, nil, nil, nil, nil]` is required?

Not sure if the `raise` is correct here but don't want to disable RuboCop here, please rewrite the code instead of disabling the cop.

Reply:
Ah! I'll fiddle with the factoring. I did
```
raise unless cached_location.exist? && fresh
```
b/c Rubocop wanted me not to use `&&` in an `unless`
but if I do
```
raise if !(cached_location.exist? && fresh)
```
Rubocop wants me to switch `if !` to `unless`
I could do
```
raise if !cached_location.exist? || !fresh
```
but that feels hardest to read.

I'll play with it!

Reply:
@MikeMcQuaid, I just noticed your suggestion at the top of this 🤦‍♂️. I accepted your suggestions and tested them locally — they look good 👍 

I have no qualms merging
```ruby
raise if !cached_location.exist? || !fresh
```
but I wanted to have a go at making it more readable.

What do you think of this refactor? https://github.com/boblail/brew/pull/284/files (Feel free to merge or close.)

Reply:
> What do you think of this refactor? https://github.com/boblail/brew/pull/284/files (Feel free to merge or close.)

Looks great! I have ✅ but don't have permissions to merge.

Reply:
Awesome, thanks! I'm going to squash commits now on this branch.